schema/messaging.proto

```diff
@@ -0,0 +1,482 @@
+      delete: "/v1alpha3/{name=rooms/*}"
+    };
+    option (google.api.method_signature) = {
+      fields: ["name"]
+    };
+  }
+
+  // Lists all chat rooms.
+  rpc ListRooms(ListRoomsRequest) returns (ListRoomsResponse) {
+    option (google.api.http) = {
+      get: "/v1alpha3/rooms"
+    };
+  }
+
+  // Creates a blurb. If the parent field corresponds to a RoomBlurb
+  // resource, the blurb is understood to be a message in a chatroom. If the
+  // parent field is a ProfileBlurb resource name, the blurb is understood
+  // to be a post on a user profile.
+  rpc CreateBlurb(CreateBlurbRequest) returns (Blurb) {
+    option (google.api.http) = {
+      post: "/v1alpha3/{parent=rooms/*}/blurbs"
+      body: "*"
+      additional_bindings: {
+        post: "v1alpha3/{parent=users/*/profile}/blurbs"
+        body: "*"
+      }
+    };
+    option (google.api.method_signature) = {
+      fields: ["parent", "blurb.text", "blurb.user"]
+    };
+    option (google.api.method_signature) = {
+      fields: ["parent", "blurb.image", "blurb.user"]
+    };
+  }
+
+  // Retrieves the Blurb with the given resource name.
+  rpc GetBlurb(GetBlurbRequest) returns (Blurb) {
+    option (google.api.http) = {
+      get: "/v1alpha3/{name=rooms/*/blurbs/*}"
+      additional_bindings: {
+        get: "v1alpha3/{name=users/*/profile/blurbs/*}"
+      }
+    };
+    option (google.api.method_signature) = {
+      fields: ["name"]
+    };
+  }
+
+  // Updates a blurb.
+  rpc UpdateBlurb(UpdateBlurbRequest) returns (Blurb) {
+    option (google.api.http) = {
+      patch: "/v1alpha3/{blurb.name=rooms/*/blurbs/*}"
+      body: "*"
+      additional_bindings: {
+        patch: "v1alpha3/{blurb.name=users/*/profile/blurbs/*}"
+        body: "*"
+      }
+    };
+  }
+
+  // Deletes a blurb.
+  rpc DeleteBlurb(DeleteBlurbRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/v1alpha3/{name=rooms/*/blurbs/*}"
+      additional_bindings: {
+        delete: "v1alpha3/{name=users/*/profile/blurbs/*}"
+      }
+    };
+    option (google.api.method_signature) = {
+      fields: ["name"]
+    };
+  }
+
+  // Lists blurbs for a specific chat room or user profile depending on the
+  // parent resource name.
+  rpc ListBlurbs(ListBlurbsRequest) returns (ListBlurbsResponse) {
+    option (google.api.http) = {
+      get: "/v1alpha3/{parent=rooms/*}/blurbs"
+      additional_bindings: {
+        get: "v1alpha3/{parent=users/*/profile}/blurbs}"
+      }
+    };
+    option (google.api.method_signature) = {
+      fields: ["parent"]
+    };
+  }
+
+  // This method searches through all blurbs across all rooms and profiles
+  // for blurbs containing to words found in the query. Only posts that
+  // contain an exact match of a queried word will be returned.
+  rpc SearchBlurbs(SearchBlurbsRequest) returns (google.longrunning.Operation) {
+    option (google.api.http) = {
+      post: "/v1alpha3/rooms/-/blurbs:search"
```

Comment:
Optional: I think this should be `{parent=rooms/-}`, to permit expansibility later.

Reply:
Is it common add http params when they are not present in the request message?

Reply:
I am also suggesting that `string parent` should be added to the request message (if you do this).

Reply:
Ah gotcha. SGTM.